admin/src/components/ConfigsTable/index.tsx

```diff
@@ -0,0 +1,86 @@
+import pluginId from '../../pluginId';
+import Config from '../../../../types/Config';
+import {
+  Table,
+  TFooter,
+  Thead,
+  Tbody,
+  Tr,
+  Td,
+  Th,
+  Flex,
+  Box,
+  IconButton,
+} from '@strapi/design-system';
+import { Plus, Trash } from '@strapi/icons';
+import { useMemo, useState } from 'react';
+import { getFetchClient } from '@strapi/helper-plugin';
+import { Link } from 'react-router-dom';
+
+export default function ConfigsTable() {
+  const [data, setData] = useState<Config[]>([]);
+  const { get, del } = getFetchClient();
+
+  useMemo(() => {
+    get(`/${pluginId}/config`).then((res) => {
+      setData(res.data);
+    });
+  }, [setData]);
+
+  async function handleDetete(id: string) {
+    const deleteConfirm = await del(`/${pluginId}/config/${id}`);
+    if (deleteConfirm) {
+      const newData = await get(`/${pluginId}/config`);
+      setData(newData.data);
+    }
+  }
+
+  const COL_COUNT = 4;
+
+  const FooterButton = () => {
+    return (
+      <Link to={location => ({ ...location, pathname: `${location.pathname}/new-workflow` })}>
+        <TFooter icon={<Plus />}>Add another workflow</TFooter>
+      </Link>
+    );
+  };
+
+  return (
+    <Table colCount={COL_COUNT} footer={<FooterButton />}>
+      <Thead>
+        <Tr>
+          <Th>Id</Th>
+          <Th>GitHub Account</Th>
+          <Th>Repo</Th>
+          <Th>Branch</Th>
+          <Th>Workflow</Th>
+        </Tr>
+      </Thead>
+      <Tbody>
+        {data &&

```

Comment:
TBH, I am not sure how we handle loading state here? do we have any loading indicators ?

Reply:
The Loading state wasn't handled. I added a spiner in my last commit